File: Code/WebSvc.dotNet/WebSvc.dotNet/Repositories/IDb.cs
using Model;

namespace Repositories
{
    /// <summary>
    /// Interface to persistence layer
    /// </summary>
    public interface IDb
    {
        /// <summary>
        ///  Update birth date of a person in DB
        /// </summary>
        /// <param name="p"></param>
        void UpdateBirth(Person p);
    }
}

File: Code/WebSvc.Java/src/main/java/repositories/IDb.java
package repositories;

import models.Person;

/**
 * Interface to persistence layer
 */
public interface IDb {
	
	/**
	 * Old, non CLEAN
	 * @param key
	 * @param age
	 * @throws Exception
	 */
	void updateBirth(int key, int age) throws Exception;
	
	/**
	 * Update birth date of a person in DB
	 * @param p
	 * @throws Exception
	 */
	void updateBirth(Person p) throws Exception;
}
